libraries/AP_HAL_SITL/HAL_SITL_Class.cpp

```diff
@@ -74,6 +77,47 @@ HAL_SITL::HAL_SITL() :
 
 static char *new_argv[100];
 
+/*
+  save watchdog data
+ */
+static bool watchdog_save(const uint32_t *data, uint32_t nwords)
+{
+    int fd = ::open("persistent.dat", O_WRONLY|O_CREAT|O_TRUNC, 0644);
+    bool ret = false;
+    if (fd != -1) {
+        if (::write(fd, data, nwords*4) == (ssize_t)(nwords*4)) {
```

Comment:
There's a race condition here.  If you are running behind and the watchdog does kick in you may have truncated the file but not written anything to it.

I guess it's possible this could happen on `stm32` as well - worst case would be a partial update of the location so the lat/lon check against 0/0 wouldn't work correctly.  If this happens in practise I suggest staying well clear of that user.